packages/codefixes/test/codefixes.test.ts

```diff
@@ -0,0 +1,229 @@
+import { resolve } from 'node:path';
+import { describe, expect, test } from 'vitest';
+import { RehearsalService } from '@rehearsal/service';
+import { ModuleKind, ModuleResolutionKind, ScriptTarget, ImportsNotUsedAsValues } from 'typescript';
+import { TypescriptCodeFixCollection } from '../src/typescript-codefix-collection.js';
+import { getDiagnosticOrder } from '../src/get-diagnostics.js';
+import { applyCodeFix } from '../src/codefixes.js';
+import { CodeFixes } from '../src/codefixInformationMap.generated.js';
+
+describe('ts-codefixes', async () => {
+  const tsCollection = new TypescriptCodeFixCollection();
+
+  test('has test for all codefixes', async () => {
+    await Promise.all(
+      Object.entries(CodeFixes).map(async ([key, value]) => {
+        if (!value.builtIn) {
+          try {
+            expect(
+              (await fs.stat(resolve(`./test/fixtures/ts-codefixes/${key}`))).isDirectory()
+            ).toBe(true);
+          } catch (e) {
+            expect(`Missing test for ${key}.`).toBe('Not Missing');
+          }
+        }
+      })
+    );
+  });
+
+  async function runTest(
+    failingPath: string,
+    passingPath: string,
+    sidecarFiles: string[] = []
+  ): Promise<void> {
+    const input = resolve(`./test/fixtures/ts-codefixes/${failingPath}`);
+    const expectation = resolve(`./test/fixtures/ts-codefixes/${passingPath}`);
+    const service = new RehearsalService(
+      {
+        strict: true,
+        module: ModuleKind.NodeNext,
+        target: ScriptTarget.ESNext,
+        moduleResolution: ModuleResolutionKind.NodeNext,
+        importsNotUsedAsValues: ImportsNotUsedAsValues.Error,
+        useUnknownInCatchVariables: true,
+        allowUnusedLabels: false,
+        experimentalDecorators: true,
+        isolatedModules: true,
+        noImplicitOverride: true,
+      },
+      [input, ...sidecarFiles.map((file) => resolve(`./test/fixtures/ts-codefixes/${file}`))]
+    );
+    const ls = service.getLanguageService();
+    const program = ls.getProgram()!;
+    const checker = program.getTypeChecker();
+    const diagnostics = getDiagnosticOrder(service.getDiagnostics(input));
+
+    expect(diagnostics.length).toBeGreaterThanOrEqual(1);
+
+    for (const diagnostic of diagnostics) {
+      const fixes = tsCollection.getFixesForDiagnostic(
+        {
+          ...diagnostic,
+          service: ls,
+          program,
+          checker,
+        },
+        { safeFixes: true, strictTyping: true }
+      );
+
+      for (const fix of fixes) {
+        applyCodeFix(fix, {
+          getText(filename: string): string {
+            return service.getFileText(filename);
+          },
+          setText(filename: string, text: string) {
+            service.setFileText(filename, text);
+          },
+        });
+      }
+    }
+
+    expect(service.getFileText(input)).toEqual(await fs.readFile(expectation, 'utf-8'));
+  }
+
+  test('addMissingAync', async () => {
```

Comment:
You can use `test.each` or even `describe.each` instead of create test for every codefix
```
const test = {
  'convertToTypeOnlyImport': [
    'convertToTypeOnlyImport/failing/fail-1.ts',
    'convertToTypeOnlyImport/passing/pass-1.ts',
     ['convertToTypeOnlyImport/sample-1.ts']
  ],,

test.each(tests)(...
```

Reply:
You can even generate `const codefixesTests =` from files automatically... through it looks too magic :)